crates/oxc_allocator/src/vec2/mod.rs

```diff
@@ -2822,7 +2822,7 @@ impl<'a, 'bump, T> Drain<'a, 'bump, T> {
     unsafe fn move_tail(&mut self, extra_capacity: usize) {
         let vec = self.vec.as_mut();
         let used_capacity = self.tail_start + self.tail_len;
-        vec.buf.reserve(used_capacity, extra_capacity);
+        vec.buf.reserve(usize_to_u32(used_capacity), usize_to_u32(extra_capacity));
```

Comment:
I'm not sure if this is sound or not. Are `used_capacity ` and `extra_capacity` definitely `<= u32::MAX`? `Drain` is hard to wrap your head around!

Reply:
It is sound!
https://github.com/oxc-project/oxc/blob/7c5d8afddafdebdca2591547290959a5ee25941c/crates/oxc_allocator/src/vec2/mod.rs#L2843-L2845